src/java.desktop/share/classes/javax/swing/plaf/synth/SynthGraphicsUtils.java

```diff
@@ -731,4 +744,47 @@ public int getIconHeight() {
             return synthIcon.getIconHeight(context);
         }
     }
+
+    /*
+     * Repaints all the components with the mnemonics in the given window and all its owned windows.
+     */
+    static void repaintMnemonicsInWindow(final Window w) {
+        if (w == null || !w.isShowing()) {
+            return;
+        }
+
+        final Window[] ownedWindows = w.getOwnedWindows();
+        for (final Window element : ownedWindows) {
+            repaintMnemonicsInWindow(element);
+        }
+
+        repaintMnemonicsInContainer(w);
+    }
+
+    /*
+     * Repaints all the components with the mnemonics in container.
+     * Recursively searches for all the subcomponents.
+     */
+    static void repaintMnemonicsInContainer(final Container cont) {
+        for (int i = 0; i < cont.getComponentCount(); i++) {
```

Comment:
i think you can use `final Component c : cont.getComponents()` here the same way you did in the repaintMnemonicsInWindow method

Reply:
Updated.